src/ch9329.cpp

```diff
@@ -207,6 +235,120 @@ ch9329::mouse_relative& ch9329::mouse_relative::right_button(bool p_pressed)
   return *this;
 }
 
+// keyboard media functions
+ch9329::keyboard_media& ch9329::keyboard_media::press_media_key(media_key p_key)
+{
+  auto byte_num = (static_cast<uint8_t>(p_key) >> 3) + 1;
+  auto bit_num = static_cast<uint8_t>(p_key) & 0b111;
+
+  switch (bit_num) {
+    case 0:
+      m_media_data[byte_num] = m_media_data[byte_num] | 0b1;
+      break;
+    case 1:
+      m_media_data[byte_num] = m_media_data[byte_num] | 0b10;
+      break;
+    case 2:
+      m_media_data[byte_num] = m_media_data[byte_num] | 0b100;
+      break;
+    case 3:
+      m_media_data[byte_num] = m_media_data[byte_num] | 0b1000;
+      break;
+    case 4:
+      m_media_data[byte_num] = m_media_data[byte_num] | 0b10000;
+      break;
+    case 5:
+      m_media_data[byte_num] = m_media_data[byte_num] | 0b100000;
+      break;
+    case 6:
+      m_media_data[byte_num] = m_media_data[byte_num] | 0b1000000;
+      break;
+    case 7:
+      m_media_data[byte_num] = m_media_data[byte_num] | 0b10000000;
+      break;
+  }
```

Comment:
```suggestion
      m_media_data[byte_num] |= 1 << bit_num;
```

You can eliminate the large switch case with just a single bit shift and or expression.